packages/backend-app-api/src/services/implementations/httpAuth/httpAuthServiceFactory.ts

```diff
@@ -41,128 +39,205 @@ function getTokenFromRequest(req: Request) {
     const matches = authHeader.match(/^Bearer[ ]+(\S+)$/i);
     const token = matches?.[1];
     if (token) {
-      return { token, isCookie: false };
+      return token;
     }
   }
 
+  return undefined;
+}
+
+function getCookieFromRequest(req: Request) {
   const cookieHeader = req.headers.cookie;
   if (cookieHeader) {
     const cookies = parseCookie(cookieHeader);
     const token = cookies[BACKSTAGE_AUTH_COOKIE];
     if (token) {
-      return { token, isCookie: true };
+      return token;
     }
   }
 
-  return { token: undefined, isCookie: false };
+  return undefined;
 }
 
 const credentialsSymbol = Symbol('backstage-credentials');
+const limitedCredentialsSymbol = Symbol('backstage-limited-credentials');
 
 type RequestWithCredentials = Request & {
   [credentialsSymbol]?: Promise<BackstageCredentials>;
+  [limitedCredentialsSymbol]?: Promise<BackstageCredentials>;
 };
 
 class DefaultHttpAuthService implements HttpAuthService {
+  readonly #auth: AuthService;
+  readonly #discovery: DiscoveryService;
+  readonly #pluginId: string;
+
   constructor(
-    private readonly auth: AuthService,
-    private readonly discovery: DiscoveryService,
-    private readonly pluginId: string,
-  ) {}
+    auth: AuthService,
+    discovery: DiscoveryService,
+    pluginId: string,
+  ) {
+    this.#auth = auth;
+    this.#discovery = discovery;
+    this.#pluginId = pluginId;
+  }
 
   async #extractCredentialsFromRequest(req: Request) {
-    const { token, isCookie } = getTokenFromRequest(req);
+    const token = getTokenFromRequest(req);
     if (!token) {
-      return createCredentialsWithNonePrincipal();
+      return await this.#auth.getNoneCredentials();
     }
 
-    const credentials = toInternalBackstageCredentials(
-      await this.auth.authenticate(token),
-    );
-    if (isCookie) {
-      if (credentials.principal.type !== 'user') {
-        throw new AuthenticationError(
-          'Refusing to authenticate non-user principal with cookie auth',
-        );
-      }
-      credentials.authMethod = 'cookie';
+    return await this.#auth.authenticate(token);
+  }
+
+  async #extractLimitedCredentialsFromRequest(req: Request) {
+    const token = getTokenFromRequest(req);
+    if (token) {
+      return await this.#auth.authenticate(token, {
+        allowLimitedAccess: true,
+      });
+    }
+
+    const cookie = getCookieFromRequest(req);
+    if (!cookie) {
```

Comment:
would like to invert this to be uniform with the above

Reply:
yep! 👍